esp_sysview/src/include/esp_sysview_heap_trace_module.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * SPDX-FileCopyrightText: 2018-2025 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdarg.h>
+#include "esp_err.h"
+#include "SEGGER_RTT.h" // SEGGER_RTT_ESP_Flush
+
+/* Forward declaration for esp_trace integration */
+typedef struct esp_trace_encoder esp_trace_encoder_t;
+
+/**
+ * @brief Inject encoder handle to SEGGER RTT layer.
+ *
+ * This maintains proper architectural layering. SEGGER RTT can accesses it's transport
+ *
+ * @param encoder  Pointer to encoder instance from esp_trace
+ *
+ * @return 0 if successful, -1 if encoder is not initialized or missing required functions.
+ */
+int SEGGER_RTT_ESP_SetEncoder(esp_trace_encoder_t *encoder);
+
+/**
+ * @brief Get the encoder handle for accessing transport functions.
+ *
+ * This is used by SEGGER_SYSVIEW_Config_FreeRTOS.c to access transport lock functions.
+ *
+ * @return Pointer to encoder instance, or NULL if not initialized.
+ */
+esp_trace_encoder_t *SEGGER_SYSVIEW_GetEncoder(void);
+
+/**
+ * @brief Flushes remaining data in SystemView trace buffer to host.
+ *
+ * @return ESP_OK.
+ */
+static inline esp_err_t esp_sysview_flush(void)
+{
+    SEGGER_RTT_ESP_Flush();
```

Comment:
My 2 cents, can we hide it in the .c file? then we don't need to expose the "SEGGER_RTT" to the user.